interfaces/repo.go

```diff
@@ -764,6 +764,47 @@ func (r *Repository) connected(snapName, plugOrSlotName string) ([]*ConnRef, err
 	return conns, nil
 }
 
+// ConnectionsForDeviceKey returns all hotplug connections for given device key and interface name.
+func (r *Repository) ConnectionsForDeviceKey(deviceKey, ifaceName string) ([]*ConnRef, error) {
```

Comment:
Following up on points in base PRs: s/DeviceKey/HotplugKey/

Equivalent in docs above it and in parameter name.

Also, per reviews in base PRs, the order should generally be:

_(ifaceName, deviceKey)_

since the key is built by the interface itself, and so only makes inside the context of the given interface.